### ISSUE TITLE
Integral Steps: Wrong indication but exact output

### ISSUE THREAD

**Issue by cosmoscalibur** [sympy/sympy_gamma#86]
Using Sympy Gamma for [`integrate(2/(log(n)**2*n), (n,2, oo))`](http://gamma.sympy.org/input/?i=integrate%282%2F%28log%28n%29**2*n%29%2C+%28n%2C2%2C+oo%29%29), one substitution is wrong and incomplete (mixing new and old variable) and then a repeated variable appear without reason in next step (still without complete substitution).

**Comment by lidavidm** [comment]
It looks like SymPy 1.0 doesn't show this behavior, but the version used by Gamma does. We just need to update the SymPy version: https://github.com/sympy/sympy_gamma/issues/81

**Comment by cosmoscalibur** [comment]
Steps change but persistent some strange behaviors (see 1.A.I.)

**Comment by abhinov2** [comment]
Can i add this whole thing in my website as a problem solver for student.
getting steps module.
If yes , how can i do it

**Comment by lidavidm** [comment]
@abhinov2 Yes, of course, so long as you follow the license (MIT). Take a look at https://github.com/sympy/sympy_gamma/blob/master/app/logic/stepprinter.py

**Comment by abhinov2** [comment]
Q1.why can't i find sympy.integral.manualintegrate in sympy module ?
Q2.I am going through the file :why are u referring it , any special reason.
Q3.I am new at python , can i get documentation about the linkages between the modules of sympgamma.

**Comment by lidavidm** [comment]
1. It is [there][manualintegrate], but you have to import it manually.
2. That is the file which actually takes the steps and formats it for display, the "steps module" if you will. If by "whole thing" you meant Gamma itself (sorry, I should have read more carefully) then you will want to look at all the source files. Gamma is a Django project, but I don't know much about Django, and can't tell you how it can be integrated.
3. Unfortunately we don't have documentation for Gamma itself. You will want to familiarize yourself with SymPy which is documented (http://docs.sympy.org) and with Django.

[manualintegrate]: https://github.com/sympy/sympy/blob/master/sympy/integrals/manualintegrate.py

**Comment by abhinov2** [comment]
On git i found it ,but can not find it under integral documentation neither in the module index  -" (http://docs.sympy.org) " .
2.Can you give us some reference from where we can get some help regrading integration.

**Comment by lidavidm** [comment]
Sorry, it's not really documented. If you have a specific question about how it works, please reach out to me. It's a rather naive algorithm that simply tries various rules straight out of a high school calculus course.

To clarify - did you want more documentation on how integration works in SymPy, or how the steps works?

**Comment by abhinov2** [comment]
Can you come on skype someday, so you can help me out with this python code of sympy.
I really want to know the portion of code from where that text box is coming on sympygamma site.
Actually , i can not find the links betwen the codes.

And yes i want more documentation for integration works in SymPy and how steps works?

**Comment by abhinov2** [comment]
I just want to know the starting point ,
Because i want to integrate this whole module in my website , so i want to know about every corner of the code.

**Comment by lidavidm** [comment]
Sorry, I'm not available - I have a full time job. The text is generated by the module I initially pointed you to. For integration, the official SymPy docs are your best bet.

**Comment by cosmoscalibur** [comment]
After all this, what about the true bug? Query still indicate wrong steps.

**Comment by lidavidm** [comment]
@cosmoscalibur sorry to spam you all. We need to update the SymPy version, there are some PRs to do so, but it seems like they've stalled.

**Comment by abhinov2** [comment]
@lidavidm hello,
I want to understand after taking input from index.html , where that input is going, help me out with this,

![captureinput](https://user-images.githubusercontent.com/15181017/28725238-86e53ae0-73da-11e7-9867-f9c5ff898de1.JPG)

**Comment by abhinov2** [comment]
@lidavidm I have inserted image , I want to understand where this input is going . Code written between <form>    </form> , I am asking .

**Comment by cosmoscalibur** [comment]
@abhinov2, this issue is about another topic (a bug). You can create a new issue for your requests.

**Comment by lidavidm** [comment]
Hi @abhinov2 could you please create a new issue, or email me directly: li.davidm96@gmail.com.

**Comment by cosmoscalibur** [comment]
@lidavidm some strange steps in the last version. Error is not the same but steps are still wrong.
